mobly/controllers/android_device_lib/snippet_client_v2.py

```diff
@@ -446,9 +446,27 @@ def create_socket_connection(self):
       self.log.debug(
           'Failed to connect to localhost, trying 127.0.0.1: %s', str(err)
       )
-      self._conn = socket.create_connection(
-          ('127.0.0.1', self.host_port), _SOCKET_CONNECTION_TIMEOUT
-      )
+      try:
+        self._conn = socket.create_connection(
+            ('127.0.0.1', self.host_port), _SOCKET_CONNECTION_TIMEOUT
+        )
+      except ConnectionRefusedError as err2:
+        _, stdout, _ = utils.run_command(
+            f'netstat tulpn | grep ":{self.host_port}"', shell=True
+        )
+        if not stdout:
```

Comment:
why is the check `not stdout` instead of checking `stderr`?

Reply:
`stderr` is not empty when the execution succeds. The is because it is not running with `sudo` so `stderr` contains permission warning.

Thus I'm checking `stdout`.

Reply:
Right, using the ret code would be slightly better. Modified.